support/globalconfig/network.go

```diff
@@ -20,12 +23,21 @@ func NetworkConfig() *configv1.Network {
 	}
 }
 
-func ReconcileNetworkConfig(cfg *configv1.Network, hcp *hyperv1.HostedControlPlane) {
+func ReconcileNetworkConfig(cfg *configv1.Network, hcp *hyperv1.HostedControlPlane) error {
 	var clusterNetwork []configv1.ClusterNetworkEntry
 	for _, entry := range hcp.Spec.Networking.ClusterNetwork {
 		hostPrefix := uint32(entry.HostPrefix)
 		if hostPrefix == 0 {
-			hostPrefix = defaultHostPrefix
+			ipv4, err := util.IsIPv4(entry.CIDR.String())
```

Comment:
but we might add a comment here e.g. `this should never happen because UnmarshalJSON for IPNet type is explicitly invoking ParseCIDR over the input so API machinery would reject the CR`